src/classes/UTIL_ACCT_Naming.cls

```diff
@@ -35,296 +35,647 @@
-        String hhAccNamingFormat = UTIL_CustomSettingsFacade.getSettings().Household_Account_Naming_Format__c;
-        if (hhAccNamingFormat == Label.acctNamingOther) {
-            hhAccNamingFormat = UTIL_CustomSettingsFacade.getSettings().Household_Other_Name_Setting__c;
+
+    private static List<Contact> sortContacts(List<Contact> cons) {
+        List<UTIL_SortContacts> sortedContacts = new List<UTIL_SortContacts>();
+        List<Contact> finalSortedContacts = new List<Contact>();
+
+        for (Contact c : cons) {
+            sortedContacts.add(new UTIL_SortContacts(c));
         }
-        if (hhAccNamingFormat!= null) {
-            return buildAccountName(cons, hhAccNamingFormat);
-        } else {
-            Set<String> householdLastNames = new Set<String>();
-            for (Contact con : cons) {
+        sortedContacts.sort();
+
+        for (UTIL_SortContacts us : sortedContacts) {
+            finalSortedContacts.add(us.contactRecord);
+        }
+
+        return finalSortedContacts;
+    }
+
+    /*******************************************************************************************************
+    * @description This sets the default Account name for each respective Account model.
+    * @param cons a list of Contacts whose Account needs Name update.
+    * @param accountNamingFormat is a string of the Account name format currently set in the org.
+    * @return String.
+    */
+    private static String defaultAccountName(List<Contact> cons, String accountNamingFormat) {
+        Set<String> householdLastNames = new Set<String>();
+        String finalAccountName;
+
+        for (Contact con : cons) {
+            if (defaultRecTypeId == hhAccountRecordTypeId) {
                 householdLastNames.add(con.LastName);
+            } else if (defaultRecTypeId == adminAccountRecordTypeId) {
+                finalAccountName = con.LastName + ' ' + System.label.DefaultAdminName;
+                break;
+            } else {
+                finalAccountName = con.LastName + ' ' + System.label.DefaultAccountName;
+                break;
             }
+        }
+
+        if (householdLastNames.size() > 0) {
             String name = '';
+            String strAnd = ' ' + Label.defaultNamingConnector + ' ';
             Integer counter = 0;
+
             for (String lastName : householdLastNames) {
                 name += lastName;
                 counter++;
                 if (counter < householdLastNames.size()) {
-                    name += ' ' + Label.defaultNamingConnector +' ';
+                    name += strAnd;
                 }
             }
-            return name + ' ' + System.label.DefaultHouseholdName;
+
+            if (accountNamingFormat == hhNameFormat) {
+                finalAccountName =  name + ' ' + System.label.DefaultHouseholdName;
+            }
         }
+
+        return finalAccountName;
     }
 
+    private static String chLToken = '{!';
```

Comment:
Maybe move these 3 variables to the top of the class with the others? I had a hard time finding these at first